### PR TITLE
feat(tool): add google custom search tool

### PR DESCRIPTION
**Pull request by abughali**
### Description

- Adding Google Custom Search Tool
- Updated Readme

### Checklist

- [x] I have read the [contributor guide](https://github.com/i-am-bee/bee-agent-framework/blob/main/CONTRIBUTING.md)
- [x] Linting passes: `yarn lint` or `yarn lint:fix`
- [x] Formatting is applied: `yarn format` or `yarn format:fix`
- [x] Unit tests pass: `yarn test:unit`
- [x] E2E tests pass: `yarn test:e2e`
- [x] Tests are included 
- [x] Documentation is changed or added
- [x] Commit messages and PR title follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary)

### CODE REVIEW

**Comment by abughali** [comment]
@Tomas2D FYK review

File: src/tools/search/googleCustomSearch.ts [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    const startIndex = (page - 1) * this.options.maxResultsPerPage! + 1;
+
+    const response = await this.client.cse.list({
+      cx: this.options.cseId,
+      q: input,
+      auth: this.options.apiKey,
+      num: this.options.maxResultsPerPage,
+      start: startIndex,
+      safe: "active",
+    });
```
Comment (Tomas2D) [inline src/tools/search/googleCustomSearch.ts:142]: Are results somehow sorted?

**Review by Tomas2D — CHANGES_REQUESTED** [review]

File: package.json [hunk 2]
```diff
@@ -112,6 +112,7 @@
     "zod-to-json-schema": "^3.23.3"
   },
   "peerDependencies": {
+    "@googleapis/customsearch": "*",
```
Comment (Tomas2D) [inline package.json:115]: Users should use the version that we developed our extension against. So you might use `^3.2.0` or `^3.0.0` (but test it first if you decide to go with 3.0.0)

File: src/tools/search/googleCustomSearch.ts [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+        .describe(
+          `Search result page (each page contains maximally ${this.options.maxResultsPerPage} results)`,
+        )
+        .default(1)
+        .optional(),
+    });
+  }
+
+  protected apiKey: string | undefined;
+  protected cseId: string | undefined;
```
Comment (Tomas2D) [inline src/tools/search/googleCustomSearch.ts:89]: I think you can change it to

```
  protected apiKey: string;
  protected cseId: string;
```

**Review by Tomas2D — APPROVED** [review]
Great work 🚀

**Pull request merged**